From 8ceba3d2be95c5e93b6a69ec6deb73ef6c6838c7 Mon Sep 17 00:00:00 2001
From: Markus Boehme <markus.boehme@chainguard.dev>
Date: Tue, 9 Dec 2025 21:02:49 +0000
Subject: [PATCH 4/5] vendor: cilium/ebpf: use non-crypto rand for kprobe trace
 group

The cilium/ebpf package uses crypto/rand to come up with a best-effort
unique name for a new kprobe trace group. This use is not relevant to
security and can be switched over to plain math/rand/v2 to avoid having
to pull in the crypto package.

Signed-off-by: Markus Boehme <markus.boehme@chainguard.dev>
---
 vendor/github.com/cilium/ebpf/internal/tracefs/kprobe.go | 7 ++-----
 1 file changed, 2 insertions(+), 5 deletions(-)

diff --git a/vendor/github.com/cilium/ebpf/internal/tracefs/kprobe.go b/vendor/github.com/cilium/ebpf/internal/tracefs/kprobe.go
index bdadf4d8..96d09e0b 100644
--- a/vendor/github.com/cilium/ebpf/internal/tracefs/kprobe.go
+++ b/vendor/github.com/cilium/ebpf/internal/tracefs/kprobe.go
@@ -1,9 +1,9 @@
 package tracefs
 
 import (
-	"crypto/rand"
 	"errors"
 	"fmt"
+	"math/rand/v2"
 	"os"
 	"path/filepath"
 	"runtime"
@@ -57,10 +57,7 @@ func RandomGroup(prefix string) (string, error) {
 		return "", fmt.Errorf("prefix '%s' must be alphanumeric or underscore: %w", prefix, ErrInvalidInput)
 	}
 
-	b := make([]byte, 8)
-	if _, err := rand.Read(b); err != nil {
-		return "", fmt.Errorf("reading random bytes: %w", err)
-	}
+	b := rand.Uint64()
 
 	group := fmt.Sprintf("%s_%x", prefix, b)
 	if len(group) > 63 {
-- 
2.51.2

